### ISSUE TITLE
Localization of Mathematics

### ISSUE THREAD

**Issue by jakubkaczor** [typst/typst#3238]
### Description

I am not sure whether formulas are considered text, and whether `text`'s function argument `lang` should be considered, but I wish there was localization for mathematics.
1. In polish, there is a convention to typeset displayed integrals with limits, but it is not the default after setting `text`'s `lang` parameter.
2. Also, the "less than or equal to" sign's slanted variant should be used. Of course, the symbol `le.eq.slant` could be used directly, but it would be nice to have `<=` be interpreted for it by default. Correct me if I am wrong, but for now `<=` alias cannot be changed, even manually.
Here is an exemplar code showing the difference.
```typst
#set par(justify: true)
#set text(lang: "pl")

= How It Looks for Now

A displayed formula:
$
  integral_a^b f(x) dif x <= M.
$

An inline formula: $integral_a^b f(x) dif x$.

= How It Should Look

// This is just an example integral. Other variants should also be considered.
#show math.integral: i => math.limits(i, inline: false)

A displayed formula:
$
  integral_a^b f(x) dif x lt.eq.slant M.
$

An inline formula: $integral_a^b f(x) dif x$.
```
![The output.](https://github.com/typst/typst/assets/57760986/b765b2ca-164d-4923-aef7-95ff39130df6)

I wish the output of the second section was achievable using the source for the first section after setting `lang` to `"pl"`.

### Use Case

This feature would be useful for typesetting mathematics according to conventions in a language automatically, and more comfortably.

**Comment by xkevio** [comment]
While a nice idea, it is simply not really feasible to support every mathematical convention of every language as of right now. Even then, mathematical notation isn't really universal anyway and can change or have different meaning, probably even within the Polish language.

But, you can already replace some shorthands like `<=` or `>=` to use the slanted variant. Either use a show-rule like so:
```typ
#show sym.lt.eq: $lt.eq.slant$
#show sym.gt.eq: $gt.eq.slant$
```
or the [quick-maths](https://github.com/typst/packages/tree/main/packages/preview/quick-maths/0.1.0) package. As for the integral, you've already demonstrated an easy way to modify it. I understand the idea of having a setting that automatically adheres to conventions, but typst provides the tools necessary to workaround or even enable them with just a bit of manual work. Localization is just plain difficult -- especially for such usecases -- unfortunately.

**Comment by jakubkaczor** [comment]
> is simply not really feasible to support every mathematical convention of every language as of right now

Due to lack of features making it easy, I suspect. Apart from this, why not? Regarding `<=`, it is the same idea as with `smartquotes`.

> mathematical notation isn't really universal anyway and can change or have different meaning

That is correct, but I didn't mean localizing mathematical conventions, but typesetting conventions.

> But, you can already replace some shorthands like <= or >= to use the slanted variant. Either use a show-rule like so:
> ```
> #show sym.lt.eq: $lt.eq.slant$
> #show sym.gt.eq: $gt.eq.slant$
> ```
> or the [quick-maths](https://github.com/typst/packages/tree/main/packages/preview/quick-maths/0.1.0) package.

You didn't really change the mapping between `<=` and `sym.lt.eq`—it is the same. You changed the look of `sym.lt.eq` to `sym.lt.eq.slant`, there is a difference. Despite, I think that the `quick-maths` solution you suggested would kind of work for a manual redefinition, thanks.

> I understand the idea of having a setting that automatically adheres to conventions, but typst provides the tools necessary to workaround or even enable them with just a bit of manual work. Localization is just plain difficult -- especially for such usecases -- unfortunately.

Making something possible is important, but I thought that one advantage of Typst over LaTeX, and one of the reasons of its conception, was to provide basic things built-in. In LaTeX there are packages like babel, polyglossia, and you could wire it up yourself. I consider support for localization basic for a typesetting system which is meant to be able to typeset documents in multiple languages. As Typst already has `lang` parameter for text, and `smartquotes`, I suppose it aims to be able to typeset in multiple languages. I don't see why formulas would suppose to be an exception.

I took the information technologies course during a mathematical program at a university. One of the basic things we were supposed to change in LaTeX were the things I have just mentioned, and I suppose snippets doing it are what all of my colleagues have in every document they typeset. I hoped Typst can just have it after setting a language, as it seems to be such a common thing.

**Comment by Enivex** [comment]
> Due to lack of features making it easy, I suspect. 

No, because there's very little standardization in mathematical notation, even within a single country or language.

**Comment by jakubkaczor** [comment]
> I can almost guarantee you that if you asked Polish mathematicians, they wouldn't all agree that integrals should be typeset with limits, or that the slanted inequality should be used.

I don't see a value in this statement. I suppose for almost any convention, you can find an individual that does not agree with this convention, unless the people knowing about it is very small, e.g., 1.

xkevio, Enivex, you seem to reason on the basis of a premise that there is virtually no standardization of typesetting mathematics. In general, this is not true, but indeed some specific cases, including the ones mentioned so far, may be lacking.

There are at least the following standards.

1. ISO/IEC 10646 (Unicode). In the specific ISO/IEC 10646:2020, which is publicly available, on the page 291, you can find the entire section on mathematical operators, see page 291, that associates symbols with their names. One may argue that the standard associates names, not meanings, and I think that is correct. Nevertheless, I think following this standard is a better default, than using random symbols.
2. ISO/IEC 80000-2. This one “specifies mathematical symbols, explains their meanings, and gives verbal equivalents and applications”.
3. Polish norm PN-N-01050:1968 (Podstawowe oznaczenia matematyczne, ang. Basic mathematical notation), has been deprecated.
4. Polish norm BN-65 7440-05 (Rules of composition of mathematical formulae), has been deprecated.

In 2 there are given variants of integrals with different limits placement, but it doesn't say when to use which.

The 4 indeed gives the specification of the position of limits and many more details. Despite this fact, 3 and 4 are no longer in use. I suppose they have been deprecated because of partially conflicting with the international ISO/IEC 80000-2.

According to Wikipedia, it used to be mandatory to follow „Normy branżowe”, eg. 4, up to 1993. 4 has been admitted in 1965. Unless 4 was deprecated before 1993, throughout the period of, more or less, 28 years it was mandatory to follow the placement of limits I described. I think it is worth considering as a default for a language and to be named traditional. That does not mean that I think it definitely should. Perhaps there are some other reasons, so the rules are no longer relevant.

As a conclusion, I am not convinced that there are current, normalized rules in polish that would be worth customizing for in Typst by default, but I cannot tell for other languages. Therefore, I am not closing this issue with this comment. Should I, and should it be reopened when needed?

I believe ISO/IEC 80000-2 should be followed, but as an international standard, it has little to do with localization, the topic of this issue.
